impl/client/src/main/javascript/web/pentaho/ccc/visual/Funnel.js

```diff
@@ -0,0 +1,155 @@
+/*!
+ * Copyright 2010 - 2019 Hitachi Vantara. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "pentaho/module!_",
+  "pentaho/visual/impl/View",
+  "common-ui/echarts",
+  "./_util",
+  "pentaho/visual/color/util"
+], function (module, BaseView, echarts, util, visualColorUtils) {
+
+  "use strict";
+
+  return BaseView.extend(module.id, {
+
+    configureLabel: function (option) {
+      var font = util.defaultFont(null, 12);
+
+      var label = {
+        show: true,
+        position: this.model.labelsOption,
+        formatter: '{b}: {d}%',
+        backgroundColor: 'transparent',
+        fontSize: font.substring(0, font.indexOf(' ')),
+        fontFamily: font.substring(font.indexOf(' ') + 1)
+      };
+
+      option.series.forEach(function (row) {
+        row.label = label;
+      })
+    },
+
+    configureLegend: function (option, records) {
+      var categories = [];
+      var font = util.defaultFont(null, 14);
+
+      records.forEach(function (record) {
+          categories.push(record.name);
+        }
+      );
+
+      var legend = {
+        data: categories,
+        type: 'scroll',
+        textStyle: {
+          fontSize: font.substring(0, font.indexOf(' ')),
+          fontFamily: font.substring(font.indexOf(' ') + 1),
+        }
+      };
+      option.legend = legend;
+    },
+
+    configureColors: function () {
```

Comment:
The method does not configure, in the sense of the other methods, of causing side-effects on some data-structure stored in this instance. IMO, it could more appropriately be called `_buildColors`, for example.